cmd/ebrelayer/relayer/prophecyHandler.go

```diff
@@ -0,0 +1,193 @@
+		sub.SugaredLogger.Errorw("failed to start a sifchain client.",
+			errorMessageKey, err.Error())
+		completionEvent.Add(1)
+		go sub.Start(txFactory, completionEvent, symbolTranslator)
+		return
+	}
+
+	defer client.Stop() //nolint:errcheck
+
+	t := time.NewTicker(time.Second * wakeupTimer)
+
+	quit := make(chan os.Signal, 1)
+	signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
+	defer close(quit)
+
+	for {
+		select {
+		case <-quit:
+			sub.SugaredLogger.Warn("we receive the quit signal and exit")
+			return
+
+		case <-t.C:
+			sub.SugaredLogger.Info("timer triggered, start to check cosmos message")
+			sub.handleNewProphecyCompleted(client)
+		}
+	}
+}
+
+// Get all not processed Prophecy via rpc and handle them in batch
+func (sub CosmosSub) handleNewProphecyCompleted(client *tmClient.HTTP) {
+	// Start Ethereum client
+	ethClient, err := ethclient.Dial(sub.EthProvider)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed to connect to Ethereum node.",
+			errorMessageKey, err.Error())
+		return
+	}
+
+	cosmosBridgeAddress, err := txs.GetAddressFromBridgeRegistry(ethClient, sub.RegistryContractAddress, txs.CosmosBridge, sub.SugaredLogger)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed to get the cosmos bridge address.",
+			errorMessageKey, err.Error())
+		return
+	}
+
+	// get the last global nonce from smart contract
+	lastSubmittedNonce, err := GetLastNonceSubmitted(ethClient, cosmosBridgeAddress, sub.SugaredLogger)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed to get the last submitted nonce.",
+			errorMessageKey, err.Error())
+		return
+	}
+
+	prophecyInfoArray := GetAllProphciesCompleted(sub.TmProvider, sub.NetworkDescriptor, lastSubmittedNonce.Uint64()+1)
+
+	batches := (len(prophecyInfoArray) + 4) / 5
+	batch := 0
+
+	for batch < batches {
+		end := (batch + 1) * 5
+		if end > len(prophecyInfoArray) {
+			end = len(prophecyInfoArray)
+		}
+
+		batchProphecyInfo := prophecyInfoArray[batch*5 : end]
+
+		if !sub.handleBatchProphecyCompleted(batchProphecyInfo) {
+			sub.SugaredLogger.Errorw("fail to process prophecy.")
+			return
+		}
+
+		batch++
+	}
+}
+
+// Parses event data from the msg, event, builds a new ProphecyClaim, and relays it to Ethereum
+func (sub CosmosSub) handleBatchProphecyCompleted(
+	batchProphecyInfo []*oracletypes.ProphecyInfo) bool {
+
+	sub.SugaredLogger.Infow(
+		"handle batch prophecy completed.",
+		"prophecyInfoArray", batchProphecyInfo,
+	)
+
+	client, auth, target, err := tryInitRelayConfig(sub)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed in init relay config.",
+			errorMessageKey, err.Error())
+		return false
+	}
+
+	// Initialize CosmosBridge instance
+	cosmosBridgeInstance, err := cosmosbridge.NewCosmosBridge(target, client)
+	if err != nil {
+		sub.SugaredLogger.Errorw("failed to get cosmosBridge instance.",
+			errorMessageKey, err.Error())
+		return false
+	}
+
+	maxRetries := 5
+	i := 0
+
+	for i < maxRetries {
+		err = txs.RelayBatchProphecyCompletedToEthereum(
```

Comment:
We should only send 1 tx across the bridge at a time to start, we can move to batching in the relayer later.

Instead of calling RelayBatch, just call `RelayProphecyCompletedToEthereum` instead.

This means that we can remove the logic to turn data into batches earlier on in this function.